### PR TITLE
Remove finalize of autovirtualdc

### PR DESCRIPTION
**Pull request by yamatcha**
Remove finalize since it's not used by autovirtualdc controller.

Signed-off-by: yamatcha <soju-yamashita@cybozu.co.jp>

### CODE REVIEW

**Comment by umezawatakeshi** [comment]
> Remove finalize since it's not used by autovirtualdc controller.

Does it mean that nyamber controller will not delete the corresponding VirtualDC when an AutoVirtualDC is deleted?

**Comment by yamatcha** [comment]
> Does it mean that nyamber controller will not delete the corresponding VirtualDC when an AutoVirtualDC is deleted?

When an AutoVirtualDC was deleted, the corresponding VirtualDC is deleted by garbage collection of OwnerReference.

**Review by umezawatakeshi — APPROVED** [review]

File: controllers/autovirtualdc_controller_test.go [hunk 1]
```diff
@@ -232,6 +218,22 @@ var _ = Describe("AutoVirtualDC controller", func() {
...
+					return err
+				}).ShouldNot(HaveOccurred())
+				err = k8sClient.Delete(ctx, vdc)
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, client.ObjectKey{Name: "test-avdc", Namespace: testNamespace}, vdc)
+					return apierrors.IsNotFound(err)
+				}).Should(BeTrue())
+			}
```
Comment (kmdkuk) [inline controllers/autovirtualdc_controller_test.go:236]: Is it necessary to delete VirtualDC resources here?
All VirtualDC resources are deleted in AfterEach, so I don't think it is necessary to delete them for cleaning purposes.

Comment (yamatcha) [reply controllers/autovirtualdc_controller_test.go:236]: I want to delete after each testcase is tested, but these tests is not in 'It'.
So I wrote procedure for delete VirtualDC.

Comment (kmdkuk) [reply controllers/autovirtualdc_controller_test.go:236]: Sorry, I understand.

**Review by kmdkuk — APPROVED** [review]
LGTM

**Pull request merged**